app/assets/javascripts/components/common/article_viewer.jsx

```diff
@@ -80,15 +98,15 @@ const ArticleViewer = createReactClass({
     if (!this.state.whocolorFetched && this.isWhocolorLang()) {
       this.fetchWhocolorHtml();
     }
-    window.history.pushState({}, '', `?showArticle=${this.props.article.id}`);
+    // Add article id in the URL
+    this.addParamToURL(`showArticle=${this.props.article.id}`);
   },
 
   hideArticle(e) {
     this.setState({ showArticle: false });
     const viewer = document.getElementsByClassName('article-viewer')[0];
```

Comment:
Since this `viewer` node is only the concern of the removeParam code, I think this makes more sense to be part of `removeParamFromURL` instead of passed in as an argument.